schemes/musica/tuvx/musica_ccpp_tuvx.F90

```diff
@@ -62,31 +64,57 @@ subroutine tuvx_init(vertical_layer_dimension, vertical_interface_dimension, &
       return
     end if
 
-    radiators => radiator_map_t( error )
+    temperature_profile => create_temperature_profile( height_grid, errmsg, errcode )
+    if (errcode /= 0) then
```

Comment:
Is there a reason why `has_error_occurred` that is used in the subsequent subroutine calls is not used here?

Reply:
I don't have a definitive answer for that. My impression is that this is CCPP-enabled code that cares about error codes and may not recognize `error_t`, but I could be mistaken. Hi @mattldawson , could you clarify this?

Reply:
That's correct - in the MUSICA library, functions return an `error_t` instance that `has_error_occurred()` converts into the `errmsg` and `errcode` that CAM-SIMA uses and returns true if an error has occurred. In this case, the `create_temperature_profile()` is a function in the atmospheric physics repo, so it just returns the `errmsg` and `errcode` directly.

Reply:
Thanks for clarifying!